Fix invalid: BADSIG F60F4B3D7FA2AF80 error in basic lab 5

BC-Li opened the pull request:
## Environment
* OS: Ubuntu 18.04
* Python 3.6
## Command
`docker build -f Dockerfile.gpu -t train_dl .`
## Error
When the dockerfile.gpu runs the command `RUN conda install pytorch torchvision cudatoolkit=10.1 -c pytorch`to install `cudatookit`, because the CDN cached a newer signature Release.gpg but did NOT refresh Release's cache at the same time(see https://github.com/NVIDIA/nvidia-docker/issues/613), the installation will return an error like
```bash
W: GPG error: https://developer.download.nvidia.cn/compute/machine-learning/repos/ubuntu1804/x86_64  Release: The following signatures were invalid: BADSIG F60F4B3D7FA2AF80 cudatools <cudatools@nvidia.com>
E: The repository 'https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release' is not signed.
```
 (see also https://github.com/NVIDIA/nvidia-docker/issues/969)
The installation may fail 1000 times unless adding a proxy. (but this error may happen somewhere else according to the comments)
## Solution
I tried to change the downloading source from Nvidia to Aliyun and TUNA, (like https://github.com/NVIDIA/nvidia-docker/issues/969#issuecomment-703186192) and removed the **adding pubkey** step on my second try(to reproduce) to build the image and it worked for me.
```dockerfile
RUN curl -fsSL https://mirrors.aliyun.com/nvidia-cuda/ubuntu1804/x86_64/7fa2af80.pub | apt-key add - // the command I removed
```
>  At the first time I added all the commands in https://github.com/NVIDIA/nvidia-docker/issues/969#issuecomment-703186192 and resolved the problem, but at the second time I removed the command **above** and it still works. However I didn't know whether it is because I already added before. So if the "removed" version cause some problems, please consider readding the command above.

There is another question that I'm not quite sure whether it is a good idea to add such operations in an official tutorial because the problems such as network error may vary, but the tutorial is for everyone. Also, changing the downloading source and adding pubkey may cause some security problems. So this is just a trick to make it work in my location (in China) XD

Again, Thanks for the awesome tutorial, I learnt a lot from it!

YanjieGao commented:
Thanks BC-Li, 

1. i think let's add a 常见问题 section at this main README, and link to a new readme issue.md to explain the issue is better.
![image](https://user-images.githubusercontent.com/5576848/120259802-30ec3900-c2c7-11eb-875b-7e6725d3f4d7.png)

2. make the network fixing part as a comment then the other country people could use the default one. Users meet the same issue could refer 常见问题 part to remove the comment use the new download source.

3. make the docker file comment as English to consist with other files. 

Thanks.

BC-Li commented:
Hi @YanjieGao,

Thanks for the advice! I've added a new `issue.md` part and commented the fix in `dockerfile.gpu`.

as for the 
>3. make the docker file comment as English to consist with other files.

I found that in the original `dockerfile.gpu` the comments are all written in Chinese, like:
```dockerfile
# 创建镜像中的文件夹，用于存储新的代码或文件
RUN mkdir -p /src/app
```
So I didn't change my comment to English in order to consist with the original version. Or should I change the comments in original version of the dockerfile to English too? 


Also, the @microsoft-cla bot asked me to sign the Contributor License Agreement in `issue.md`, which I have no idea how to do in a **Markdown** file. Can you tell me how to sign it so that I can add some new commits?

Thanks!

YanjieGao commented:
https://github.com/microsoft/AI-System/pull/27

@BC-Li Hi, your branch of PR27 has been deleted by yourself which let this PR cannot be merged. Could you create a branch or a new PR?

YanjieGao reviewed (commented):
> Hi @YanjieGao,
> 
> Thanks for the advice! I've added a new `issue.md` part and commented the fix in `dockerfile.gpu`.
> 
> as for the
> 
> > 1. make the docker file comment as English to consist with other files.
> 
> I found that in the original `dockerfile.gpu` the comments are all written in Chinese, like:
> 
> ```dockerfile
> # 创建镜像中的文件夹，用于存储新的代码或文件
> RUN mkdir -p /src/app
> ```
> 
> So I didn't change my comment to English in order to consist with the original version. Or should I change the comments in original version of the dockerfile to English too?
> 
> Also, the @microsoft-cla bot asked me to sign the Contributor License Agreement in `issue.md`, which I have no idea how to do in a **Markdown** file. Can you tell me how to sign it so that I can add some new commits?
> 
> Thanks!

OK, let's keep cn

> Hi @YanjieGao,
> 
> Thanks for the advice! I've added a new `issue.md` part and commented the fix in `dockerfile.gpu`.
> 
> as for the
> 
> > 1. make the docker file comment as English to consist with other files.
> 
> I found that in the original `dockerfile.gpu` the comments are all written in Chinese, like:
> 
> ```dockerfile
> # 创建镜像中的文件夹，用于存储新的代码或文件
> RUN mkdir -p /src/app
> ```
> 
> So I didn't change my comment to English in order to consist with the original version. Or should I change the comments in original version of the dockerfile to English too?
> 
> Also, the @microsoft-cla bot asked me to sign the Contributor License Agreement in `issue.md`, which I have no idea how to do in a **Markdown** file. Can you tell me how to sign it so that I can add some new commits?
> 
> Thanks!

OK Let's keep cn with origin

YanjieGao commented on Labs/BasicLabs/Lab5/README.md:
```diff
@@ -102,6 +102,7 @@
 * [3. Docker部署PyTorch训练程序 - 内容，步骤，作业](./train.md)
 * [4. Docker部署PyTorch推理程序 - 内容，步骤，作业](./inference.md)
 * [5. 进阶学习](./extend.md)
+* [6. 常见问题](./issue.md)
```
We could make "常见问题" same level with "参考资料".

BC-Li replied:
Thanks! 
According to the advice, I plan to make "常见问题" section as a h2 title and link it to `issue.md`, like:
## [常见问题](./issue.md)
Is that OK?

YanjieGao reviewed (commented):
> Hi @YanjieGao,
> 
> Thanks for the advice! I've added a new `issue.md` part and commented the fix in `dockerfile.gpu`.
> 
> as for the
> 
> > 1. make the docker file comment as English to consist with other files.
> 
> I found that in the original `dockerfile.gpu` the comments are all written in Chinese, like:
> 
> ```dockerfile
> # 创建镜像中的文件夹，用于存储新的代码或文件
> RUN mkdir -p /src/app
> ```
> 
> So I didn't change my comment to English in order to consist with the original version. Or should I change the comments in original version of the dockerfile to English too?
> 
> Also, the @microsoft-cla bot asked me to sign the Contributor License Agreement in `issue.md`, which I have no idea how to do in a **Markdown** file. Can you tell me how to sign it so that I can add some new commits?
> 
> Thanks!

![image](https://user-images.githubusercontent.com/5576848/120420151-5eec7e80-c396-11eb-9a35-adcbbdb3db6e.png)
bocheng, could you see this button?

BC-Li commented on Labs/BasicLabs/Lab5/issue.md:
```diff
@@ -0,0 +1,13 @@
+### 原因
+NVIDIA 的 CDN 更新了新的 GPG 签名缓存 `Release.gpg`，但对应的本体 `Release` 并未更新，这会造成上述错误。
+### 解决方法
+清除在 `Dockerfile.gpu` 内注释为`# 解决网络问题` 下的命令，令其运行将下载源换为 TUNA 与 Aliyun 源，可以解决问题。
```
OK, I have updated them in the new commit.

BC-Li commented:
Hi @YanjieGao,

I doublechecked the Dockerfile.gpu in the commit and changed following things:
* Move the fix code and let it be executed after `FROM nvidia/cuda:10.1-cudnn7-devel`, which should be at the beginning of the dockerfile. 
* Comment the fix in the dockerfile.
* Add `RUN apt-get update && apt-get install -y --no-install-recommends curl` to install curl or the command `RUN curl -fsSL https://mirrors.aliyun.com/nvidia-cuda/ubuntu1804/x86_64/7fa2af80.pub | apt-key add - ` won't be executed because `curl` is not installed.
* Removed the "Changing source to TUNA" part in https://github.com/NVIDIA/nvidia-docker/issues/969#issuecomment-703186192 to minimize the code (I think changing to TUNA is only accelerating the download speed instead of solving this problem).

I tested the code in my laptop and it seems to work fine. Thanks for the advice!
